lib/src/fix.rs

```diff
@@ -302,12 +302,14 @@ pub fn fix_files(
             });
             if new_value != old_value {
                 tree_builder.set_or_remove(repo_path.clone(), new_value);
-                changes += 1;
+                has_changes = true;
             }
         }
         summary.num_checked_commits += 1;
-        if changes > 0 {
+        if has_changes {
             summary.num_fixed_commits += 1;
+        }
+        if has_changes || rewriter.parents_changed() {
```

Comment:
If we handle `has_changes` and `.parents_changed()` differently, maybe we can omit `.write_tree()`, etc. at all?

```rust
if has_changes {
    ...
} else if rewriter.parents_changed() {
    rewriter.reparent().write()?;
}
```

I'm not sure if "reparent" should be included in `rewrites`.

Reply:
Makes sense to omit `.write_tree()`. I've updated that. OTOH, maybe we should optimize `MergedTreeBuilder` so it doesn't do any writes if nothing changed.

We don't use `rewrites` for anything yet, but I think it makes sense to include reparented commits there. I would expect all rewritten commits to be included.